src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -711,13 +712,38 @@ private[snowflake] class SnowflakeSQLStatement(
 
   def isEmpty: Boolean = list.isEmpty
 
-  def execute(
-    bindVariableEnabled: Boolean
-  )(implicit conn: Connection): ResultSet =
-    if (bindVariableEnabled) executeWithBindVariable(conn)
-    else executeWithoutBindVaribale(conn)
+  private[snowflake] def executeWithoutThrow(
```

Comment:
Why not just send telemetry in `catch` instead of testing for `isDefined` for the third tuple field?